lib/internal/util.js

```diff
@@ -356,6 +356,24 @@ function getConstructorOf(obj) {
   return null;
 }
 
+/**
+ * Get the current working directory while accounting for the possibility that it has been deleted.
+ * `process.cwd()` can fail if the parent directory is deleted while the process runs.
+ * @returns {string} The current working directory or the volume root if it cannot be determined.
+ */
+function getCwdSafe() {
+  const { sep } = require('path');
```

Comment:
shouldn't this be cached at module level, since users can change it?

```suggestion
const { sep } = require('path');
/**
 * Get the current working directory while accounting for the possibility that it has been deleted.
 * `process.cwd()` can fail if the parent directory is deleted while the process runs.
 * @returns {string} The current working directory or the volume root if it cannot be determined.
 */
function getCwdSafe() {
```

Reply:
I agree with Jordan: it's called `getCwdSafe`, let's make it safe.

Reply:
Okay sure. But what is the use-case of a user changing sep and expecting things to work? Wouldn't that break heaps of stuff in node?

Reply:
I think the word “safe” in the function name causes this friction. May I recommend removing the word safe from it?

Reply:
i mean sure, but requiring path at module level seems like the default and idiomatic thing to do; requiring things in function bodies is usually only done when using a module-level caching mechanism. why require it inline here?

Reply:
Let's imagine a user has this snippet somewhere in their code:

```js
Object.defineProperty(require("node:path"), "sep", { __proto__:null, get() { throw this } });
```

It makes `getCwdSafe` throw, when the whole point of the function is to never throw.

> node will have exploded long before reaching this

Admitting this is true (I don't think it is), we should fix the other places rather than taking it as a reason for not making internals more robust 😅

Reply:
> why require it inline here?

Ack; when are we supposed to require it JIT? I thought the point of Joyee's huge PR was to reduce startup overhead (which is incurred by root-level requires).

> when the whole point of the function is to never throw

Ah, okay. I do see your point: if that get() → throw is triggered outside, it isn't triggered by the "safe" util.

Reply:
> I thought the point of Joyee's huge PR was to reduce startup overhead (which is incurred by root-level requires)

That's true for non-snapshoted modules, `node:path` is part of the snapshot so the overhead is the same at the top-level as it is inlined. My understanding is that lazy-loading modules already in the snapshot is slightly worth than no lazy-loading, but that's beside the point: we can make this function actually safe, so we should do it IMO :)

The snapshot creation is explained in https://github.com/nodejs/node/blob/4ff55ec2e18f3e0e2516a3069f750dc313f5f4ae/lib/internal/bootstrap/node.js#L7-L16

We have a test to assert that no new modules are added to the bootstrap sequence, and you can see that `node:path` is already there: https://github.com/nodejs/node/blob/41b8872cbde1b4b7ba0c8c63e2544f61d5f4d2b8/test/parallel/test-bootstrap-modules.js#L51

> when are we supposed to require it

I'd say at the top-level.